tests/validation/check_validation_test.go

```diff
@@ -0,0 +1,132 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/openfga/openfga/cmd"
+	"github.com/openfga/openfga/pkg/testfixtures/storage"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/stretchr/testify/require"
+	pb "go.buf.build/openfga/go/openfga/api/openfga/v1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	healthv1pb "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/status"
+	"gopkg.in/yaml.v3"
+)
+
+type checkValidationTests struct {
+	Tests []checkValidationTest
+}
+
+type checkValidationTest struct {
+	Name    string
+	Model   string
+	Tuples  []*pb.TupleKey
+	Request request
+	Code    int
+}
+
+type request struct {
+	Tuple            *pb.TupleKey
+	ContextualTuples []*pb.TupleKey `yaml:"contextualTuples"`
+}
+
+func TestCheckValidation(t *testing.T) {
+	data, err := os.ReadFile("check_validation_test.yaml")
+	require.NoError(t, err)
+
+	var tests checkValidationTests
+	err = yaml.Unmarshal(data, &tests)
+	require.NoError(t, err)
+
+	container := storage.RunDatastoreTestContainer(t, "memory")
+	cfg := cmd.MustDefaultConfigWithRandomPorts()
+	cfg.Datastore.URI = container.GetConnectionURI()
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		if err := cmd.RunServer(ctx, cfg); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	require.NoError(t, err)
+	defer conn.Close()
+
+	// Ensure the service is up before continuing.
+	client := healthv1pb.NewHealthClient(conn)
+	policy := backoff.NewExponentialBackOff()
+	policy.MaxElapsedTime = 10 * time.Second
+	err = backoff.Retry(func() error {
+		resp, err := client.Check(ctx, &healthv1pb.HealthCheckRequest{
+			Service: pb.OpenFGAService_ServiceDesc.ServiceName,
+		})
+		if err != nil {
+			return err
+		}
+
+		if resp.GetStatus() != healthv1pb.HealthCheckResponse_SERVING {
+			return fmt.Errorf("not serving")
+		}
+
+		return nil
+	}, policy)
+	require.NoError(t, err)
+
+	runTests(t, pb.NewOpenFGAServiceClient(conn), tests)
+
+	// Shutdown the server.
+	cancel()
+}
+
+func runTests(t *testing.T, client pb.OpenFGAServiceClient, tests checkValidationTests) {
+	ctx := context.Background()
+
+	for _, test := range tests.Tests {
+		t.Run(test.Name, func(t *testing.T) {
+			resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: test.Name})
+			require.NoError(t, err)
+
+			storeID := resp.GetId()
+
+			fmt.Println(parser.MustParse(test.Model))
+			_, err = client.WriteAuthorizationModel(ctx, &pb.WriteAuthorizationModelRequest{
+				StoreId:         storeID,
+				SchemaVersion:   typesystem.SchemaVersion1_1,
+				TypeDefinitions: parser.MustParse(test.Model),
+			})
+			require.NoError(t, err)
+
+			for _, tuple := range test.Tuples {
```

Comment:
Similar comment above. We don't have any tuples to write if we are testing request validation errors, yeah?

Reply:
Yep! Thanks.